sematic/resolvers/local_resolver.py

```diff
@@ -193,13 +197,25 @@ def _future_did_resolve(self, future: AbstractFuture) -> None:
     def _future_did_fail(self, failed_future: AbstractFuture) -> None:
         super()._future_did_fail(failed_future)
 
-        run = self._get_run(failed_future.id)
+        run = api_client.get_run(failed_future.id)
 
         run.future_state = failed_future.state
 
         # We do not propagate exceptions to parent runs
-        if failed_future.state == FutureState.FAILED:
+        logger.error(
+            "Processing failure of run %s, state: %s",
+            failed_future.id,
+            failed_future.state,
+        )
+        if failed_future.state == FutureState.FAILED and run.exception is None:
             run.exception = format_exception_for_run()
+        if failed_future.state == FutureState.NESTED_FAILED and run.exception is None:
```

Comment:
This technically isn't needed for this PR, but as I was looking at some failures during development this was helpful to have.